token-account-based/chaincode-go/chaincode/token_contract.go

```diff
@@ -0,0 +1,173 @@
+package chaincode
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+// SmartContract provides functions for transferring tokens between accounts
+type SmartContract struct {
+	contractapi.Contract
+}
+
+// Mint creates new tokens and adds them to minter's account balance
+func (s *SmartContract) Mint(ctx contractapi.TransactionContextInterface, amount int) error {
+
+	// Check minter authorization - this sample assumes Org1 is the central banker with privilege to mint new tokens
+	clientMSPID, err := ctx.GetClientIdentity().GetMSPID()
+	if err != nil {
+		return fmt.Errorf("failed to get MSPID: %v", err)
+	}
+	if clientMSPID != "Org1MSP" {
+		return fmt.Errorf("client is not authorized to mint new tokens")
+	}
+
+	// Get ID of submitting client identity
+	minter, err := ctx.GetClientIdentity().GetID()
+	if err != nil {
+		return fmt.Errorf("failed to get client id: %v", err)
+	}
+
+	if amount <= 0 {
+		return fmt.Errorf("mint amount must be a positive integer")
+	}
+
+	currentBalanceBytes, err := ctx.GetStub().GetState(minter)
+	if err != nil {
+		return fmt.Errorf("failed to read minter account %s from world state: %v", minter, err)
+	}
+
+	var currentBalance int
+
+	// If minter current balance doesn't yet exist, we'll create it with a current balance of 0
+	if currentBalanceBytes == nil {
+		currentBalance = 0
+	} else {
+		currentBalance, _ = strconv.Atoi(string(currentBalanceBytes))
```

Comment:
good to add error handling on Atoi return

Reply:
True but trying to mint a non integer amount generates an error so I don't think the extra check is really necessary:

`Error: endorsement failure during invoke. response: status:500 message:"Error managing parameter param0. Conversion error. Cannot convert passed value 1.5 to int" `

Reply:
I was trying to keep it as compact as possible so that it can be easily read without distracting error handling.
I've added a comment as a compromise:
// Error handling not needed since Itoa() was used when setting the account balance, guaranteeing it was an integer.